tests/metrics/test_auc.py

```diff
@@ -277,3 +279,115 @@ def test_when_duplicates_in_interactions_insufficient(
             dtype=float,
         )
         pd.testing.assert_series_equal(metric.calc_per_user(reco, interactions), expected_metric_per_user)
+
+
+class TestDebiasableAUCMetric:
+    def setup_method(self) -> None:
+        self.reco = pd.DataFrame(
+            {
+                Columns.User: [1, 2, 3, 3, 3, 4, 5, 5, 5, 5],
+                Columns.Item: [1, 2, 1, 2, 3, 1, 1, 2, 3, 5],
+                Columns.Rank: [9, 1, 1, 2, 3, 1, 3, 7, 9, 1],
+            }
+        )
+        self.interactions = pd.DataFrame(
+            {
+                Columns.User: [1, 2, 3, 3, 3, 4, 5, 5, 5, 5],
+                Columns.Item: [1, 1, 1, 2, 3, 1, 1, 2, 3, 4],
+            }
+        )
+
+    @pytest.mark.parametrize(
+        "metric, metric_debias",
+        (
+            (
+                PartialAUC(k=1, insufficient_handling=InsufficientHandling.IGNORE),
+                PartialAUC(k=1, insufficient_handling=InsufficientHandling.IGNORE, debias_config=DEBIAS_CONFIG),
+            ),
+            (
+                PartialAUC(k=3, insufficient_handling=InsufficientHandling.IGNORE),
+                PartialAUC(k=3, insufficient_handling=InsufficientHandling.IGNORE, debias_config=DEBIAS_CONFIG),
+            ),
+            (
+                PartialAUC(k=1, insufficient_handling=InsufficientHandling.EXCLUDE),
+                PartialAUC(k=1, insufficient_handling=InsufficientHandling.EXCLUDE, debias_config=DEBIAS_CONFIG),
+            ),
+            (
+                PartialAUC(k=1, insufficient_handling=InsufficientHandling.EXCLUDE),
+                PartialAUC(k=1, insufficient_handling=InsufficientHandling.EXCLUDE, debias_config=DEBIAS_CONFIG),
+            ),
+            (
+                PAP(k=1, insufficient_handling=InsufficientHandling.IGNORE),
+                PAP(k=1, insufficient_handling=InsufficientHandling.IGNORE, debias_config=DEBIAS_CONFIG),
+            ),
+            (
+                PAP(k=3, insufficient_handling=InsufficientHandling.IGNORE),
+                PAP(k=3, insufficient_handling=InsufficientHandling.IGNORE, debias_config=DEBIAS_CONFIG),
+            ),
+            (
+                PAP(k=1, insufficient_handling=InsufficientHandling.EXCLUDE),
+                PAP(k=1, insufficient_handling=InsufficientHandling.EXCLUDE, debias_config=DEBIAS_CONFIG),
+            ),
+            (
+                PAP(k=3, insufficient_handling=InsufficientHandling.EXCLUDE),
+                PAP(k=3, insufficient_handling=InsufficientHandling.EXCLUDE, debias_config=DEBIAS_CONFIG),
+            ),
+        ),
+    )
+    def test_calc(self, metric: tp.Union[PartialAUC, PAP], metric_debias: tp.Union[PartialAUC, PAP]) -> None:
```

Comment:
You don't really need to duplicate for each metric it's version with and without debias config.  
You can keep just version without config and inside the function make 
`debiased_metric = PAP(k=metric.k, insufficient_handling=metric.insufficient_handling, debias_config=DEBIAS_CONFIG)`

Reply:
(Everywhere)

Reply:
Fixed